### PR TITLE
feat(cozy-harvest-lib): Feat/success modal

### PR DESCRIPTION
**Pull request by Crash--**
- Add KonnectorSuccess to Harvest
- Add a new route /success 
- Remove the need of the context, since we query the stack to get the URL it should be fine with the custo?
- Don't use an Intent Redirect since they don't work on mobile (with UL on Android), so we use AppLinker

### CODE REVIEW

File: packages/cozy-harvest-lib/src/components/KonnectorSuccess.jsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+import Button from 'cozy-ui/react/Button'
+import withLocales from './hoc/withLocales'
+import DriveLink from '../components/KonnectorConfiguration/Success/DriveLink'
+import BanksLink from '../components/KonnectorConfiguration/Success/BanksLink'
+import connectingIllu from '../assets/connecting-data-in-progress.svg'
+import Markdown from './Markdown'
+
+const SuccessImage = () => (
+  <img src={connectingIllu} className={'u-w-4 u-h-4'} />
```
Comment (drazik) [inline packages/cozy-harvest-lib/src/components/KonnectorSuccess.jsx]:
```suggestion
  <img src={connectingIllu} className="u-w-4 u-h-4" />
```

File: packages/cozy-harvest-lib/src/helpers/apps.js [hunk 2]
```diff
@@ -1,29 +1,46 @@
...
+  return `${storeUrl}#/discover/${app.slug}/install`
 }
 
+/**
+ *
+ * @param {Array} apps Array of apps returned by /apps /konnectors
+ * @param {Object} wantedApp io.cozy.app with at least a slug
+ * @return {Object} The io.cozy.app is installed or undefined if not
+ */
+export const isAppIsInstalled = (apps = [], wantedApp = {}) => {
```
Comment (drazik) [inline packages/cozy-harvest-lib/src/helpers/apps.js]:
```suggestion
export const isAppInstalled = (apps = [], wantedApp = {}) => {
```

File: packages/cozy-harvest-lib/src/components/KonnectorSuccess.jsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        app.predicate(this.state, this.props)
+      ),
+      app => -app.priority
+    )
+
+    const hasLinks = relatedApps.length > 0
+
+    return (
+      <ModalContent>
+        <div className={'u-ta-center'}>
```
Comment (drazik) [inline packages/cozy-harvest-lib/src/components/KonnectorSuccess.jsx]:
```suggestion
        <div className="u-ta-center">
```

File: packages/cozy-harvest-lib/src/components/KonnectorSuccess.jsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+          <SuccessImage />
+          <DescriptionContent
+            title={t('account.success.title')}
+            message={t('account.success.connect')}
+          >
+            {hasLinks && (
+              <p className={'u-mv-half'}>
```
Comment (drazik) [inline packages/cozy-harvest-lib/src/components/KonnectorSuccess.jsx]:
```suggestion
              <p className="u-mv-half">
```

File: packages/cozy-harvest-lib/src/components/KonnectorConfiguration/Success/BanksLink.jsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    if (apps.fetchStatus === 'loaded') {
+      const banksApp = {
+        slug: 'banks'
+      }
+      let url = ''
+      const banksInstalled = isAppInstalled(apps.data, banksApp)
+      if (banksInstalled) {
+        url = getUrlForApp(banksInstalled)
+      } else {
+        url = getStoreUrltoInstallAnApp(apps.data, banksApp)
```
Comment (ptbrowne) [inline packages/cozy-harvest-lib/src/components/KonnectorConfiguration/Success/BanksLink.jsx]: Nit: Using a namespace would benefit readability. store.getInstallationURL

File: packages/cozy-harvest-lib/src/components/KonnectorConfiguration/Success/DriveLink.jsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  render() {
+    const { folderId, driveQuery, t } = this.props
+    let hasDrive = false
+    if (driveQuery.fetchStatus === 'loaded') {
+      if (driveQuery.data.length > 0) {
+        hasDrive = true
+      }
+    }
+    const href =
+      hasDrive && `${driveQuery.data[0].links.related}#/files/${folderId}`
```
Comment (ptbrowne) [inline packages/cozy-harvest-lib/src/components/KonnectorConfiguration/Success/DriveLink.jsx]:
```suggestion
      hasDrive ? `${driveQuery.data[0].links.related}#/files/${folderId}` : null
```

This has the benefit of removing the condition in ButtonLink prop.

File: packages/cozy-harvest-lib/src/components/KonnectorSuccess.jsx [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+  state = {
+    trigger: null
+  }
+  componentDidMount() {
+    const { accountId, accounts } = this.props
+    const matchingTrigger = get(
+      accounts.find(account => account.account._id === accountId),
+      'trigger'
+    )
+    this.setState({ trigger: matchingTrigger })
```
Comment (ptbrowne) [inline packages/cozy-harvest-lib/src/components/KonnectorSuccess.jsx]: Should it be derived state ? If the account rendered in this component changes the trigger will be stale.

File: packages/cozy-harvest-lib/src/helpers/apps.js [hunk 8]
```diff
@@ -1,29 +1,46 @@
...
+ *
+ * @param {Object} app io.cozy.app object
+ * @return {String} url to the app
+ */
+export const getUrlForApp = app => {
+  return app.links && app.links.related
+}
 export default {
-  getStoreUrltoUpdateKonnector
+  getStoreUrltoInstallAnApp
```
Comment (ptbrowne) [inline packages/cozy-harvest-lib/src/helpers/apps.js]: apps.getStoreInstallationURL

File: packages/cozy-harvest-lib/test/components/KonnectorSuccess.spec.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    expect(root.find(DriveLink).length).toBe(0)
+  })
+
+  it('should show drive if trigger has a folder_to_save', () => {
+    setup({ folder_to_save: 'jjj' })
+    expect(root.find(DriveLink).length).toBe(1)
+  })
+
+  it('should show banks if connector has datatypes with bankAccounts', () => {
+    setup({ isBanksKonnector: true })
```
Comment (ptbrowne) [inline packages/cozy-harvest-lib/test/components/KonnectorSuccess.spec.js]: isBankingKonnector

**Pull request merged**